cpp/examples/abm_minimal.cpp

```diff
@@ -19,46 +19,21 @@
 */
 #include "abm/abm.h"
 #include "abm/household.h"
-#include <cstdio>
 #include <fstream>
 #include <string>
 #include <iostream>
 
-void write_results_to_file(const mio::abm::Simulation& sim)
-{
-    // The results are saved in a table with 9 rows.
-    // The first row is t = time, the others correspond to the number of people with a certain infection state at this time:
-    // S = Susceptible, E = Exposed, I_NS = InfectedNoSymptoms, I_Sy = InfectedSymptoms, I_Sev = InfectedSevere,
-    // I_Crit = InfectedCritical, R = Recovered, D = Dead
-    std::ofstream myfile("abm_minimal.txt", std::ios::out);
-    myfile << "# t S E I_NS I_Sy I_Sev I_Crit R D\n";
-    for (auto i = 0; i < sim.get_result().get_num_time_points(); ++i) {
-        myfile << sim.get_result().get_time(i);
-        auto v = sim.get_result().get_value(i);
-        for (auto j = 0; j < v.size(); ++j) {
-            myfile << v[j];
-            if (j < v.size() - 1) {
-                myfile << " ";
-            }
-        }
-        if (i < sim.get_result().get_num_time_points() - 1) {
-            myfile << "\n";
-        }
-    }
-    std::cout << "Results written to abm_minimal.txt" << std::endl;
-}
-
 int main()
 {
     // This is a minimal example with children and adults < 60 year old.
-    // We divided them into 3 different age groups, which are defined as follows:
+    // We divided them into 4 different age groups, which are defined as follows:
     size_t NUM_AGE_GROUPS         = 4;
     const auto AGE_GROUP_0_TO_4   = mio::AgeGroup(NUM_AGE_GROUPS - 4);
     const auto AGE_GROUP_5_TO_14  = mio::AgeGroup(NUM_AGE_GROUPS - 3);
     const auto AGE_GROUP_15_TO_34 = mio::AgeGroup(NUM_AGE_GROUPS - 2);
     const auto AGE_GROUP_35_TO_59 = mio::AgeGroup(NUM_AGE_GROUPS - 1);
```

Comment:
Usually the all-caps style is only used for macros or maybe enums, I think we should change this to num_age_groups, age_group_0_to_4, etc..

Also, still do not get this' num - x' notation, is there a reason why we use this? It is only ever used in a model setup, anywhere else in code or comments we use e.g. `AgeGroup(1)`.
I prefer explicitly writing `AgeGroup(0)`, `AgeGroup(1)`, ... since it is shorter, clearer what it does, and it seperates the indices from the strict upper bound `AgeGroup(num_age_groups)`.

Reply:
i'm with you on that for now, for the moment i will leave it as is. please discuss in abm meeting on tuesday

Reply:
It was introduced by Martin, but was not an optimal way to do it. The idea was to show more explicitly where an error resides if a user chooses `num_age_groups` unequal to 4 here. 
We can certainly have a small discussion on this.

Reply:
I just spoke with @mknaranja, we agreed that changing this to AgeGroup(0) etc. is fine.
To make sure that a user does not specify an AgeGroup >= num_agegroups, we should add some asserts when a user has to provide an AgeGroup as argument - do you @xsaschako @khoanguyen-dev  know in which methods this would make the most sense?

Reply:
Also, what do you think of renaming the AgeGroups to lower-case letters?

Reply:
Technically these are only local variables. So we can easily change them here without affecting any other file, and open a new issue to make the naming of age groups more uniform (i.e. un-caps all other files). Ok?

Reply:
ok,will do